node-notes-cli/app.js

```diff
@@ -0,0 +1,157 @@
+import { writeFile } from 'node:fs/promises';
+import { readFileSync } from 'node:fs';
+
+// Set up function and variables to handle user inputs
+const action = process.argv[2];
+const arg3 = process.argv[3];
+const arg4 = process.argv[4];
+const legalActions = ['read', 'create', 'update', 'delete'];
+
+function interpretInputs() {
+  switch (action) {
+    case 'read':
+      readNotes();
+      break;
+    case 'create':
+      newNote();
+      break;
+    case 'update':
+      updateNote();
+      break;
+    case 'delete':
+      deleteNote();
+      break;
+    default:
+      console.error(action, 'is not an acceptable action. Please try again with:', legalActions);
+      process.exit(1);
+  }
+}
+
+/**
+ * Reusable functions
+ */
+
+// Catch error handling
+function handleError(err) {
+  console.error(err);
+  process.exit(1);
+}
+
+// Evaluate ID
+async function evaluateId() {
+  try {
+    const notes = await getJournal('notes');
+    const validIds = Object.keys(notes);
+
+    if (typeof Number(arg3) !== 'number' || !validIds.includes(arg3)) {
+      console.error(arg3, 'is not a valid ID. Please provide a valid ID:', validIds.join(', '));
+      process.exit(1);
+    }
+  } catch (err) {
+    handleError();
+  }
+}
+
+// Multiple notes
+function tooManyNotes() {
+  console.error('Please note only one note at a time can be inputted. Try again and ensure your note is enclosed in quotation marks.');
+  process.exit(1);
+}
+
+// Journal retrieval
+async function getJournal(item) {
+  try {
+    const journal = await readFileSync('data.json', 'utf8');
```

Comment:
`readFileSync` is a synchronous call, so `await` is useless here. (TypeScript would warn you of that.)

The best fix, though, is to stop using the `Sync` versions of the `fs` package. Those are there for convenience, but since they block code execution while the file is read or written, they provide performance bottlenecks. So there are almost no cases where you should use them.

In this app, the user has to wait for it to complete, anyway, so you could argue that `Sync` is OK. That would be true, but I would like you to get in the habit of never using `Sync`. (I did come across one instance in my years of writing Node code where I needed it, so it's not an absolute rule, but it's definitely a 99.9999% rule!)